### PR TITLE
Iae 32618/reset all

### PR DESCRIPTION
**Pull request by ascreven-gsa**
## Motivation and Context
https://cm-jira.usa.gov/browse/IAE-32618

## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [x] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Which browsers have you tested?
- [x] Internet Explorer 11
- [x] Edge
- [x] Chrome
- [x] Firefox
- [x] Safari

## Checklist:

- [x] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [x] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/packages/sam-formly/src/lib/formly-filters/sds-filters.component.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.sds-filters__reset {
+  text-align: right;
```
Comment (davereed) [inline libs/packages/sam-formly/src/lib/formly-filters/sds-filters.component.scss:2]: lets use the util class for this?

File: libs/packages/sam-formly/src/lib/formly-filters/sds-filters.component.ts [hunk 2]
```diff
@@ -56,6 +49,11 @@ export class SdsFiltersComponent implements OnInit {
    */
   @Input() public options: FormlyFormOptions = {};
 
+  /**
+   *   Display Reset All button that returns form fields to default states -- default false.
+   */
+  @Input() public resetAll: boolean = false;
```
Comment (davereed) [inline libs/packages/sam-formly/src/lib/formly-filters/sds-filters.component.ts:55]: Per Christy, this should be always enabled. So it should always be set to true by default, and we may end up removing this option completely.

File: libs/packages/sam-formly/src/lib/formly-reset/formly-reset.component.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  width: initial;
+  padding-bottom: 0;
+  text-decoration: none;
+  &:hover, &:active {
+    background: transparent;
+    background: initial;
+    color: initial;
+    text-decoration: underline;
+  }
+}
```
Comment (davereed) [inline libs/packages/sam-formly/src/lib/formly-reset/formly-reset.component.scss:12]: We don't want to give them the ability to change the appearance of the reset all.

File: libs/packages/sam-formly/src/lib/formly-reset/formly-reset.component.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  /**
+   * Formly form options
+   */
+  @Input() options: FormlyFormOptions;
+
+  /**
+   * Pass in classes for reset button -- default .usa-button .usa-button--unstyled
+   */
+  @Input() classes: string[] = ['usa-button', 'usa-button--unstyled'];
```
Comment (davereed) [inline libs/packages/sam-formly/src/lib/formly-reset/formly-reset.component.ts:19]: We don't want to give them the ability to change the appearance of the reset all.

**Comment by ascreven-gsa** [comment]
As discussed saving dynamic styles so we can reuse component with tests elsewhere. Other changes have been addressed

File: libs/packages/sam-formly/src/lib/formly-filters/sds-filters.component.ts [hunk 5]
```diff
@@ -40,9 +40,15 @@ export class SdsFiltersComponent implements OnInit {
   /**
    *    Options for the form.
    */
-  @Input() public options: FormlyFormOptions;
+  @Input() public options: FormlyFormOptions = {};
 
   /**
+   *   Display Reset All button that returns form fields to default states -- default false.
+   */
+  @Input() public resetAll: boolean = true;
```
Comment (davereed) [inline libs/packages/sam-formly/src/lib/formly-filters/sds-filters.component.ts:48]: There should be no option to remove the reset button.

File: libs/packages/sam-formly/src/lib/formly-filters/sds-filters.component.html [hunk 6]
```diff
@@ -1,4 +1,11 @@
 <formly-form [form]="form" [fields]="fields" [options]="options" [model]="model" (modelChange)="onModelChange($event)">
 </formly-form>
-<sds-advanced-filters *ngIf="advancedFilters" [form]="form" [fields]="fields" [options]="options" [model]="model">
-</sds-advanced-filters>
+<div class="grid-row">
+  <div *ngIf="advancedFilters" class="grid-col">
+    <sds-advanced-filters [form]="form" [fields]="fields" [options]="options" [model]="model">
+    </sds-advanced-filters>
+  </div>
+  <div *ngIf="resetAll" class="grid-col text-right">
```
Comment (davereed) [inline libs/packages/sam-formly/src/lib/formly-filters/sds-filters.component.html:8]: Remove the optional resetAll.

File: libs/sam-formly-sample/src/lib/feature/filter-wrapper/filter-wrapper.component.html [hunk 7]
```diff
@@ -1,10 +1,10 @@
 <div>
     <sds-accordion-item expanded="true">
         <sds-accordion-item-header> Filters </sds-accordion-item-header>
-        <sds-filters [form]="form" [fields]="fields" [isHistoryEnable]="true" [model]="model" (filterChange)="filterChange$.next($event)"></sds-filters>
+        <sds-filters [resetAll]="true" [form]="form" [fields]="fields" [isHistoryEnable]="true" [model]="model" (filterChange)="filterChange$.next($event)"></sds-filters>
```
Comment (davereed) [inline libs/sam-formly-sample/src/lib/feature/filter-wrapper/filter-wrapper.component.html:4]: Remove resetAll

**Pull request merged**